File: Novels/packages/5-mobile-ios/StoryReader/Model/SystemList.swift
import Foundation

// SystemList model - Predefined lists (Favorites, To Read, etc.)
struct SystemList: Codable, Identifiable {
    let id: String
    let userId: String
    let listType: SystemListType
    let libraryItems: [LibrarySystemListItem]?
    let createdAt: Date
    let updatedAt: Date
    
    enum SystemListType: String, Codable {
        case favorites
        case toRead = "to-read"
        case haveRead = "have-read"
        case currentlyReading = "currently-reading"
        case recentlyAdded = "recently-added"
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case listType = "list_type"
        case libraryItems = "library_items"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

// LibrarySystemListItem model
struct LibrarySystemListItem: Codable, Identifiable {
    let id: String
    let libraryId: String
    let systemListId: String
    let createdAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case libraryId = "library_id"
        case systemListId = "system_list_id"
        case createdAt = "created_at"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/SystemList.kt
package com.storysphere.storyreader.model

data class SystemList(
    val id: String,
    val userId: String,
    val type: SystemListType,
    val storyIds: List<String> = emptyList(),
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val lastSyncedAt: Long? = null
)

enum class SystemListType {
    FAVORITES,          // Favorite stories
    TO_READ,            // Stories to read later
    HAVE_READ,          // Completed stories
    CURRENTLY_READING,  // Currently reading
    RECENTLY_ADDED      // Recently added to library
}
